api/globalSearch/index.ts

```diff
@@ -53,17 +60,19 @@ export async function getSearchResults(req: EnhancedRequest, res: Response, next
  * @param jurisdictions
  * @returns
  */
-export function generateServices(jurisdictions: Jurisdiction[]): GlobalSearchService[] {
+export function generateServices(refDataHMCTS: RefDataHMCTSService[]): GlobalSearchService[] {
   // Retrieve global search services id from config
   const globalSearchServiceIds = getConfigValue(GLOBAL_SEARCH_SERVICES);
   const globalSearchServiceIdsArray = globalSearchServiceIds.split(',');
 
   // Generate global search services
   const globalSearchServices: GlobalSearchService[] = [];
   globalSearchServiceIdsArray.forEach((serviceId) => {
-    const jurisdiction = jurisdictions ? jurisdictions.find((x) => x.id === serviceId) : null;
+    // search for the service name based on the globalSearchServiceId
+    const jurisdiction = refDataHMCTS?.length > 0 ? refDataHMCTS.filter((x) => x.ccd_service_name?.toLowerCase() === serviceId.toLowerCase()) : null;
     if (jurisdiction) {
-      globalSearchServices.push({ serviceId: jurisdiction.id, serviceName: jurisdiction.name });
+      // handle Civil service which has different service_short_description
+      globalSearchServices.push({ serviceId: jurisdiction[0].ccd_service_name, serviceName: jurisdiction.length > 1 ? 'Civil' : jurisdiction[0].service_short_description });
```

Comment:
Not quite what I wanted, but close.
I think what I wanted was
if (jurisdiction.length > 1) {
   globalSearchServices.push({ serviceId: jurisdiction[0].ccd_service_name, serviceName: jurisdiction[0].toTitleCase(jurisdiction[0].serviceName));
} else {
   globalSearchServices.push({ serviceId: jurisdiction[0].ccd_service_name, serviceName: jurisdiction[0].service_short_description);

}
function toTitleCase(str: string): string {
str.toLowerCase().replace(/\b\w/g, s => s.toUpperCase());

Reply:
I have implemented as above but the above regex wont work with all letter with CAPITAL so changed accoringly so it should convert from text from CAPITAL to Capital